### PR TITLE
[4.0] [installation] [mysql] - null date 1000-01-01 00:00:00

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
null date changed to 1000-01-01 00:00:00 for mysql 

### Testing Instructions
apply and install on mysql/pdo

### Expected result
work as before

### Documentation Changes Required
maybe here https://docs.joomla.org/Joomla_and_MySQL_8 ?

### CODE REVIEW

**Comment by richard67** [comment]
@alikon Doesn't it need to do the same changes in the existing schema updates, too, at least in those for 4.0.0?

**Comment by alikon** [comment]
sounds reasonable
done

File: administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-06-03.sql [hunk 1]
```diff
@@ -8,11 +8,11 @@ CREATE TABLE IF NOT EXISTS `#__csp` (
...
-  `modified` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
+  `created` datetime NOT NULL DEFAULT '1000-01-01 00:00:00',
+  `modified` datetime NOT NULL DEFAULT '1000-01-01 00:00:00',
   `published` tinyint(1) NOT NULL DEFAULT 0,
   PRIMARY KEY (`id`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 DEFAULT COLLATE=utf8mb4_unicode_ci;
 
 INSERT INTO `#__extensions` (`package_id`, `name`, `type`, `element`, `folder`, `client_id`, `enabled`, `access`, `protected`, `manifest_cache`, `params`, `checked_out`, `checked_out_time`, `ordering`, `state`) VALUES
-(0, 'com_csp', 'component', 'com_csp', '', 1, 1, 1, 0, '', '{}', 0, '0000-00-00 00:00:00', 0, 0);
+(0, 'com_csp', 'component', 'com_csp', '', 1, 1, 1, 0, '', '{}', 0, '1000-01-01 00:00:00', '1000-01-01 00:00:00', 0);
```
Comment (Quy) [inline administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-06-03.sql]:
```suggestion
(0, 'com_csp', 'component', 'com_csp', '', 1, 1, 1, 0, '', '{}', 0, '1000-01-01 00:00:00', 0, 0);
```

**Review by richard67 — COMMENTED** [review]
Changes are complete for joomla.sql, sample_testing.sql and the existing `4.0.0-*.sql` schema updates. What remains to be done is to check if it needs additional, new schema updates for columns not handled by the existing `4.0.0-*.sql` schema updates, so any pre-4.0 (e.g. 3.9) will have the right default values, too, after updating to 4.0.

**Comment by alikon** [comment]
btw, outside the scope of this silly PR + we still need to full manage null as a date value all over the core code

**Comment by wilsonge** [comment]
We've already agreed to convert basically all the instances you are touching here to actual null. (REF: https://github.com/joomla/joomla-cms/pull/21901 or https://github.com/joomla/joomla-cms/pull/25360)

I don't plan on merging this until we've converted all core to use null where it makes sense to (e.g. checked out times and modified at times) and then can see what individual items are remaining

**Comment by richard67** [comment]
@wilsonge The problem is: The log of Hannes' new system tests for MySQL 8 shows that the '0000-00-00 00:00:00' dates are a problem. And there is rumor in issue here and in forums why we fail on MySQL 8. Maybe this PR here will be a faster help with that than the real nullable dates?

I agree with you in general, this PR here is useless if all of these columns will be made really nullable. But will we have that soon enough? Really all of them?

Or maybe test and merge this one here soon as a quick fix before the reaul stuff comes?

If so, we should wait until Hannes' PR #25615 is merged. Then @alikon could update with latest changes from 4.0-dev and solve conflicts, and then we see in the log of Hannes' new tests if the date thing was the only problem or if we have more and which ones maybe.

**Comment by wilsonge** [comment]
I understand that it doesn't work in MySQL 8. I think you'll find the same thing in MySQL 5.7 too actually. It's not really any more work to create PRs like the two I linked to as do this. And it's possible there might be some null date's left over at the end in which case we will still need this. But it won't be as widely used as this

**Pull request closed without merging**